mirascope/core/base/_extract.py

```diff
@@ -107,8 +114,12 @@ async def inner_async(
                     fn=fn, **create_decorator_kwargs
                 )(*args, **kwargs)
                 json_output = get_json_output(call_response, json_mode)
+                fn_args = get_fn_args(fn, args, kwargs)
+                fields_from_call_args = {c: fn_args[c] for c in call_args_field_names}
```

Comment:
why not just merge these? for example:

```python
fields_from_call_args = get_fields_from_get_args(response_model, fn, args, kwargs)
```

in this example, `get_fields_from_get_args` would replace `get_call_args_field_names_and_validate`

Reply:
This is also not merged for reasons of other comments